### PR TITLE
Make pkg.install more resilient

### PR DESCRIPTION
**Pull request by MalloZup**
# Description:

- fix partially: https://trello.com/c/rp9qZp6B/761-retry-package-installation-if-it-fails

- Add retry for pkg.install: 3 times with interval 15, values inline with other states

# todos:

- [x] pkg, changelog

### CODE REVIEW

**Review by arbulu89 — COMMENTED** [review]
Hi @MalloZup , so far this would change our issues during installation.

Anyway, I was wondering if there is some kind of hack to update these variables (retry, attempts, interval) default values... This way we woundn't need to add these every pkg installation.

If not, we can go forward with this solution, but to check would be good in my opinion

**Comment by MalloZup** [comment]
@arbulu89 afaik there isn't such API currently.

IMHO we can merge this safely now. The rest is seperate refactor.

Thx for your review.

# Additional note:

I have update spec and changelog, let me know if it is ok or I should do other pkg operation tia :dancer:

**Comment by MalloZup** [comment]
I might need to test this. I was trying to playing around with `vagrant-libvirt` but  I had some issue with the installation of the vagrant-libvirt due to bundler conflitcs on leap15 :/

Is there any other way to test this without vagrant perhaps?

The only test would just be in case I added some space in yaml, this are really annoying to detect by human eyes :grin:

**Comment by arbulu89** [comment]
@MalloZup The spec and changes files are correctly changed. This way, the package will be automatically submitted to our factory OBS repository.

About the testing part, we run the tests for this project in our jenkins instance

**Review by arbulu89 — APPROVED** [review]
It looks good to me!
If in the future we find a better solution we will take care of it!

File: cluster/ntp.sls [hunk 1]
```diff
@@ -3,6 +3,9 @@
 {% if grains['osmajorrelease']|int == 12 %}
 ntp:
   pkg.installed
```
Comment (arbulu89) [inline cluster/ntp.sls]: I think colons are missing here.
Let's recheck after the tests

**Review by arbulu89 — CHANGES_REQUESTED** [review]
I have noticed a potential issue with colons. Let's recheck after this is tested

File: cluster/ntp.sls (hunk 1 above)
Comment (MalloZup) [reply cluster/ntp.sls]: nope, no colons are needed there

Comment (MalloZup) [reply cluster/ntp.sls]: yes, it might be that here are needed!..

**Comment by MalloZup** [comment]
@arbulu89 thx adressed. From your pov is there another way that I could test the formula without vagrant?

**Comment by arbulu89** [comment]
> @arbulu89 thx adressed. From your pov is there another way that I could test the formula without vagrant?

As commented in other channel, we could initially just add a yaml sanity check, at least, to avoid these issues

**Review by arbulu89 — APPROVED** [review]
LGTM!

**Pull request merged**